Tests fail if there's _ (underscore) defined in the notebook

Is there a way to handle this? 

Test:
```
from testbook import testbook

@testbook('nb.ipynb', execute=True)
def test_foo(tb):
    foo = tb.ref("foo")
    assert foo(2) == 3
```

Notebook:
```
_ = 20

def foo(x):
    return x + 1
```

Result:
```

    @testbook('nb.ipynb', execute=True)
    def test_foo(tb):
        foo = tb.ref("foo")
>       assert foo(2) == 3
E       AssertionError: assert 20 == 3
E        +  where 20 = <[TypeError('__repr__ returned non-string (type int)') raised in repr()] TestbookObjectReference object at 0x7ff04d735c90>(2)

tb.py:6: AssertionError
======================================================================== short test summary info ========================================================================
FAILED tb.py::test_foo - AssertionError: assert 20 == 3
```

--- a/testbook/client.py
+++ b/testbook/client.py
@@ -263,13 +263,14 @@ class TestbookNotebookClient(NotebookClient):
         save_varname = random_varname()
 
         inject_code = f"""
-            {save_varname} = _
-
             import json
-            json.dumps(_)
-
+            from IPython import get_ipython
             from IPython.display import JSON
-            JSON({{"value" : _}})
+
+            {save_varname} = get_ipython().last_execution_result.result
+
+            json.dumps({save_varname})
+            JSON({{"value" : {save_varname}}})
         """
 
         try:

--- a/testbook/reference.py
+++ b/testbook/reference.py
@@ -15,7 +15,7 @@ class TestbookObjectReference:
         return self.tb.value(f"type({self.name}).__name__")
 
     def __repr__(self):
-        return self.tb.value(f"repr({self.name})")
+        return repr(self.tb.value(f"repr({self.name})"))
 
     def __getattr__(self, name):
         if self.tb.value(f"hasattr({self.name}, '{name}')"):

--- a/testbook/tests/test_execute.py
+++ b/testbook/tests/test_execute.py
@@ -74,3 +74,22 @@ def test_testbook_slice_raises_error():
 def test_raise_exception(tb):
     with pytest.raises(TestbookRuntimeError):
         tb.ref("raise_my_exception")()
+
+
+@testbook('testbook/tests/resources/inject.ipynb')
+def test_underscore(tb):
+    tb.inject(
+        """
+        _ = 20
+
+        def foo(x):
+            return x + 1
+    """,
+        run=False,
+    )
+
+    tb.execute()
+
+    foo = tb.ref("foo")
+
+    assert foo(2) == 3

--- a/testbook/tests/test_reference.py
+++ b/testbook/tests/test_reference.py
@@ -12,7 +12,7 @@ def notebook():
 
 def test_create_reference(notebook):
     a = notebook.ref("a")
-    assert repr(a) == "[1, 2, 3]"
+    assert repr(a) == "'[1, 2, 3]'"
     assert a._type == 'list'
     assert a.resolve() == [1, 2, 3]
 
@@ -48,7 +48,7 @@ def test_reference(notebook):
     # a reference to that object instead
     f = Foo('bar')
 
-    assert repr(f) == "<Foo value='bar'>"
+    assert repr(f) == "\"<Foo value='bar'>\""
 
     # Valid attribute access
     assert f.say_hello

